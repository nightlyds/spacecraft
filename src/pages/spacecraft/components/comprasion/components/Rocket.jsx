import React from "react";
import PropTypes from "prop-types";

const Rocket = ({
  rocketName,
  costs,
  firstFlight,
  country,
  successRate,
  height,
  diameter,
  mass,
  engines,
  enginesVersion,
  image,
}) => {
  return (
    <div className="comprasion-rocket">
      <div className="comprasion-rocket-image-box">
        <img src={image} alt="Rocket" className="comprasion-rocket-image" />
      </div>
      <div className="comprasion-rocket-title-box">
        <div className="comprasion-rocket-detail-box">
          <h2 className="comprasion-rocket-title">{rocketName}</h2>
        </div>
      </div>
      <div className="comprasion-rocket-info">
        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{firstFlight}</span>
        </div>
        <div className="comprasion-rocket-detail-box">
          <span className="comprasion-rocket-detail">{country}</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{successRate}</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{costs} $</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{engines}</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{enginesVersion}</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{height} m</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{diameter} m</span>
        </div>

        <div className="comprasion-rocket-detail-box">
          {" "}
          <span className="comprasion-rocket-detail">{mass} kg</span>
        </div>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocketName: PropTypes.string,
  costs: PropTypes.number,
  firstFlight: PropTypes.string,
  country: PropTypes.string,
  successRate: PropTypes.number,
  height: PropTypes.number,
  diameter: PropTypes.number,
  mass: PropTypes.number,
  engines: PropTypes.string,
  enginesVersion: PropTypes.string,
  image: PropTypes.string,
};

export default Rocket;
