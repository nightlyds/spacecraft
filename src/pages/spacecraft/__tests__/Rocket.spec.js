import React from "react";
import Rocket from "../components/comprasion/components/Rocket.jsx";
import { shallow } from "enzyme";

const setUp = (props) => shallow(<Rocket {...props} />);

describe("Rocket component test", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("Snapshot test", () => {
    expect(component).toMatchSnapshot();
  });
});
