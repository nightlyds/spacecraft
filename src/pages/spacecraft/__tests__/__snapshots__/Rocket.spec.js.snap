// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component test Snapshot test 1`] = `
<div
  className="comprasion-rocket"
>
  <div
    className="comprasion-rocket-image-box"
  >
    <img
      alt="Rocket"
      className="comprasion-rocket-image"
    />
  </div>
  <div
    className="comprasion-rocket-title-box"
  >
    <div
      className="comprasion-rocket-detail-box"
    >
      <h2
        className="comprasion-rocket-title"
      />
    </div>
  </div>
  <div
    className="comprasion-rocket-info"
  >
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      />
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
      <span
        className="comprasion-rocket-detail"
      />
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      />
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      >
         $
      </span>
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      />
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      />
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      >
         m
      </span>
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      >
         m
      </span>
    </div>
    <div
      className="comprasion-rocket-detail-box"
    >
       
      <span
        className="comprasion-rocket-detail"
      >
         kg
      </span>
    </div>
  </div>
</div>
`;
